### PR TITLE
feat: add support for adding session_id without user and password

### PR DESCRIPTION
**Pull request by ryosagisu**
My company will disable password authentication and move every API call to use JWT authentication.
We already developed logic to get `session_id` value from JWT login.

These changes will remove `metabase_user` and `metabase_password` as required arguments. 
And adding validation to check if user provided either of `session_id` or both `metabase_user` and `metabase_password`.

### CODE REVIEW

File: dbtmetabase/models/exceptions.py [hunk 1]
```diff
@@ -6,6 +6,10 @@ class NoDbtSchemaSupplied(Exception):
     """Thrown when using folder parser without supplying a schema"""
 
 
+class NoMetabaseCredentialsOrSessionIdSupplied(Exception):
```
Comment (gouline) [inline dbtmetabase/models/exceptions.py:9]: Session ID is a type of credential too, so should just be `NoMetabaseCredentialsSupplied`

Comment (ryosagisu) [reply dbtmetabase/models/exceptions.py]: updated

File: dbtmetabase/models/interface.py [hunk 2]
```diff
@@ -84,6 +89,11 @@ def prepare_metabase_client(self, dbt_models: Optional[List[MetabaseModel]] = No
         if dbt_models is None:
             dbt_models = []
 
+        if not (self.user and self.password) and not self.session_id:
+            raise NoMetabaseCredentialsSupplied(
+                f"Credentials or Session ID is not supplied"
```
Comment (gouline) [inline dbtmetabase/models/interface.py:94]:
```suggestion
                f"Credentials or session ID not supplied"
```

File: dbtmetabase/models/interface.py [hunk 3]
```diff
@@ -84,6 +89,11 @@ def prepare_metabase_client(self, dbt_models: Optional[List[MetabaseModel]] = No
         if dbt_models is None:
             dbt_models = []
 
+        if not (self.user and self.password) and not self.session_id:
+            raise NoMetabaseCredentialsSupplied(
+                f"Credentials or session ID not supplied"
```
Comment (gouline) [inline dbtmetabase/models/interface.py:94]:
```suggestion
                "Credentials or session ID not supplied"
```

File: dbtmetabase/models/interface.py [hunk 4]
```diff
@@ -19,8 +23,8 @@ def __init__(
         self,
         database: str,
         host: str,
-        user: str,
-        password: str,
+        user: Optional[str] = None,
+        password: Optional[str] = None,
```
Comment (gouline) [inline dbtmetabase/models/interface.py:27]: These type changes must be propagated down to `MetabaseClient`, that's why your CI is failing

**Review by gouline — CHANGES_REQUESTED** [review]

File: dbtmetabase/models/interface.py (hunk 4 above)
Comment (ryosagisu) [reply dbtmetabase/models/interface.py:27]: updated, should passed the CI now

**Review by gouline — APPROVED** [review]

**Pull request merged**